enterprise/models.py

```diff
@@ -2304,3 +2302,81 @@ def __repr__(self):
         Return uniquely identifying string representation.
         """
         return self.__str__()
+
+
+class PendingEnterpriseCustomerAdminUser(TimeStampedModel):
+    # pylint: disable=line-too-long
+    """
+    Model for pending enterprise admin users.
+
+    .. pii: The user_email field contains PII, but locally deleted via enterprise.signals.assign_or_delete_enterprise_admin_role when the admin registers a new account.
+    .. pii_types: email_address
+    .. pii_retirement: local_api, consumer_api
+    """  # pylint: enable=line-too-long
+
+    enterprise_customer = models.ForeignKey(
+        EnterpriseCustomer,
+        blank=False,
+        null=False,
+        on_delete=models.CASCADE,
+    )
+    user_email = models.EmailField(null=False, blank=False, unique=True)
+    history = HistoricalRecords()
+
+    class Meta:
+        app_label = 'enterprise'
+        ordering = ['created']
+        unique_together = (('enterprise_customer', 'user_email'),)
+
+    @cached_property
+    def admin_registration_url(self):
+        """
+        Returns a URL to be used by a pending enterprise admin user to register their account.
+        """
+        registration_url = '{}/{}/admin/register'.format(
+            settings.ENTERPRISE_ADMIN_PORTAL_BASE_URL,
+            self.enterprise_customer.slug
+        )
+        return registration_url
+
+    def activate_admin_permissions(self, user, role):
+        """
+        Activates admin permissions for an existing PendingEnterpriseCustomerAdminUser.
+
+        Specifically, the "enterprise_admin" system-wide role is assigned to the user and
+        the PendingEnterpriseCustomerAdminUser record is removed.
+
+        Arguments:
+            user: a User instance
+            role: a SystemWideEnterpriseRole instance
+        """
+        try:
+            pending_admin_user = PendingEnterpriseCustomerAdminUser.objects.get(
+                user_email=user.email,
+            )
```

Comment:
Couldn't there be multiple of these since user_email is unique with the customer? (This might be resolved by the question on that uniqueness)